server/src/main.rs

```diff
@@ -0,0 +1,201 @@
+        let response = Pilots {
+            pilots: PILOTS.lock().unwrap().clone(),
+        };
+        Ok(Response::new(response))
+    }
+
+    async fn pilot_by_id(&self, request: Request<Id>) -> Result<Response<Pilot>, Status> {
+        let id = request.into_inner().id;
+
+        let res = PILOTS
+            .lock()
+            .unwrap()
+            .clone()
+            .into_iter()
+            .filter(|x| x.id == id)
+            .collect::<Vec<Pilot>>();
+        if res.len() == 1 {
+            Ok(Response::new(res[0].clone()))
+        } else {
+            Err(Status::not_found("Not found"))
+        }
+    }
+
+    async fn vertiports(
+        &self,
+        _request: Request<VertiportFilter>,
+    ) -> Result<Response<Vertiports>, Status> {
+        let response = Vertiports {
+            vertiports: VERTIPORTS.lock().unwrap().clone(),
+        };
+        Ok(Response::new(response))
+    }
+
+    async fn vertiport_by_id(&self, request: Request<Id>) -> Result<Response<Vertiport>, Status> {
+        let id = request.into_inner().id;
+
+        let res = VERTIPORTS
+            .lock()
+            .unwrap()
+            .clone()
+            .into_iter()
+            .filter(|x| x.id == id)
+            .collect::<Vec<Vertiport>>();
+        if res.len() == 1 {
+            Ok(Response::new(res[0].clone()))
+        } else {
+            Err(Status::not_found("Not found"))
+        }
+    }
+
+    async fn insert_flight_plan(
+        &self,
+        request: Request<FlightPlan>,
+    ) -> Result<Response<FlightPlan>, Status> {
+        let mut fps = FLIGHT_PLANS.lock().unwrap();
+        let mut flight_plan = request.into_inner();
+        flight_plan.id = fps[fps.len() - 1].id + 1;
+        fps.push(flight_plan.clone());
+        Ok(Response::new(flight_plan))
+    }
+
+    async fn update_flight_plan_by_id(
+        &self,
+        request: Request<FlightPlan>,
+    ) -> Result<Response<FlightPlan>, Status> {
+        let flight_plan = request.into_inner();
+        let mut fps = FLIGHT_PLANS.lock().unwrap();
+        let found_fp = fps.iter_mut().find(|x| x.id == flight_plan.id);
+        if found_fp != None {
+            let mut fp = found_fp.unwrap();
+            //todo update all fields
+            fp.flight_status = flight_plan.flight_status;
+            println!("SOME {:?}", fp);
+            Ok(Response::new(fp.clone()))
+        } else {
+            Err(Status::not_found("Not found"))
+        }
+    }
+
+    /// Returns ready:true when service is available
+    async fn is_ready(
+        &self,
+        _request: Request<ReadyRequest>,
+    ) -> Result<Response<ReadyResponse>, Status> {
+        let response = ReadyResponse { ready: true };
+        Ok(Response::new(response))
+    }
+}
+
+///Main entry point: starts gRPC Server on specified address and port
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    dotenv().ok();
```

Comment:
Add a comment for what this is doing

Reply:
Did you mean  what `dotenv().ok();` is doing? I added comment for that.